internal/pkg/istio-router-check/cmd/root.go

```diff
@@ -0,0 +1,186 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/getyourguide/istio-config-validator/internal/pkg/istio-router-check/envoy"
+	"github.com/getyourguide/istio-config-validator/internal/pkg/istio-router-check/helpers"
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	"istio.io/istio/pkg/util/protomarshal"
+)
+
+const envoyRouterCheckTool string = "router_check_tool"
+
+type RootCommand struct {
+	ConfigDir               string
+	TestDir                 string
+	Details                 bool
+	DisableDeprecationCheck bool
+	OnlyShowFailures        bool
+	FailThreshold           float64
+	CoverageAll             bool
+	OutputDir               string
+	DetailedCoverage        bool
+	Verbosity               int
+}
+
+func NewCmdRoot() (*cobra.Command, error) {
+	rootCmd := &RootCommand{}
+	cmd := &cobra.Command{
+		Use: "istio-router-check",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if _, err := exec.LookPath(envoyRouterCheckTool); err != nil {
+				return fmt.Errorf("missing route table check tool: %w", err)
+			}
+			logger := slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{
+				Level: slog.Level(rootCmd.Verbosity),
+			}))
+			ctx := logr.NewContextWithSlogLogger(cmd.Context(), logger)
+			cmd.SetContext(ctx)
+			return nil
+		},
+		RunE:         rootCmd.Run,
+		SilenceUsage: true,
+	}
+	cmd.Flags().StringVarP(&rootCmd.ConfigDir, "config-dir", "c", "", "directory containing virtualservices")
```

Comment:
Is there some documentation for these flags we could link to?

Reply:
There is [this](https://www.envoyproxy.io/docs/envoy/latest/operations/tools/route_table_check_tool#install-tools-route-table-check-tool) but it is not an exact 1:1 map as our `config-dir` for example reads Istio configuration and not Envoy route. Maybe it would be better to keep our own documentation.